### PR TITLE
Quick ActorCache refactor

### PR DESCRIPTION
**Pull request by MellowYarker**
These commits were scattered before and after the larger read coalesce changes, and since that change is really big and complicated, I figured it would be easier to yank these quick refactors out into a separate PR so we can focus on the scary stuff elsewhere.

### CODE REVIEW

File: src/workerd/io/actor-cache.h
Comment (a-robinson) [inline src/workerd/io/actor-cache.h:418]: Only if we want to keep separate DIRTY and FLUSHING states. If they're going to merge, I don't see much point in having `isFoo` getters for each possible status.

File: src/workerd/io/actor-cache.h
Comment (a-robinson) [inline src/workerd/io/actor-cache.h:419]: The `KJ_FAIL_ASSERT` in `isDirty` is awkwardly encoding knowledge about its caller(s). It's nice to have that assertion there since it does narrow the space of possible states we need to worry about, but it does entangle this type with the rest of `ActorCache` a bit.

**Review by a-robinson — APPROVED** [review]
This set of changes LGTM modulo the TODO(now) in the `Make Entry::syncStatus private` commit. If you want to get it in ahead of the other PRs, feel free.

**Review by fhanau — APPROVED** [review]
LGTM

File: src/workerd/io/actor-cache.c++
Comment (justin-mp) [inline src/workerd/io/actor-cache.c++:109]: While it can make sense to have a const value parameter in a function definition, I don't think it makes sense to do so here.  Two reasons:
1. The fact that you've declared `this->options` as const means that you can only use `this->options` as const.  The `const` in the parameter declaration doesn't do anything for you here.
2. The `const` in the parameter declaration prevents you from moving the value, e.g., `options(kj::mv(options))`.

File: src/workerd/io/actor-cache.h
Comment (justin-mp) [inline src/workerd/io/actor-cache.h:845]: I don't think these const declarations buy you anything.  Two reasons:
1. You can only set the values on construction, which seems like an unreasonable constraint on users of the struct.
2. It makes this struct non-moveable.

File: src/workerd/io/limit-enforcer.h
Comment (justin-mp) [inline src/workerd/io/limit-enforcer.h:28]: This doesn't make sense because it restricts what users of the return value can do with the struct.  For example, users cannot `kj::mv` them.

File: src/workerd/io/actor-cache.h
Comment (justin-mp) [inline src/workerd/io/actor-cache.h:887]: This one makes sense.  Let's keep this.  :-)

File: src/workerd/io/actor-cache.c++
Comment (justin-mp) [inline src/workerd/io/actor-cache.c++:1]: FYI, note that KJ uses const to mean thread-safe too.  See https://github.com/capnproto/capnproto/blob/v2/style-guide.md#constness

File: src/workerd/io/actor-cache.h
Comment (justin-mp) [inline src/workerd/io/actor-cache.h:414]: nit: please put the return type here instead of `auto`.  It saves the reader trying to figure out what the return type is.

File: src/workerd/io/actor-cache.h
Comment (justin-mp) [inline src/workerd/io/actor-cache.h:654]: I love these kinds of refactors!  Making something easier to track is awesome!

File: src/workerd/io/actor-cache.h
Comment (justin-mp) [inline src/workerd/io/actor-cache.h:445]: If we have this function, but we're not supposed to use it, that's an indication of something buggy in our abstraction layers.

A better approach might be to remove `setSyncStatus` and have setClean and setDirty instead.

**Review by justin-mp — CHANGES_REQUESTED** [review]

File: src/workerd/io/actor-cache.c++
Comment (MellowYarker) [reply src/workerd/io/actor-cache.c++]: https://github.com/cloudflare/workerd/pull/1914/commits/a8da46b64c7cd1c429527277c1f1236beda426ed

Yeah, this makes sense. Thanks!

File: src/workerd/io/actor-cache.h
Comment (MellowYarker) [reply src/workerd/io/actor-cache.h]: I think I overshot a bit here :sweat_smile:, I wanted to see more `const` in the other parts of actor cache to make it easier to reason about, but that was tricky, so all that energy went into this struct https://github.com/cloudflare/workerd/pull/1914/commits/bdbae9bfac75544a05c4945c94ac33b5bd8640b1

File: src/workerd/io/actor-cache.h
Comment (MellowYarker) [reply src/workerd/io/actor-cache.h]: https://github.com/cloudflare/workerd/pull/1914/commits/fcef6f68058edfed70c7d00ebcb1efc1ae55752c

File: src/workerd/io/limit-enforcer.h
Comment (MellowYarker) [reply src/workerd/io/limit-enforcer.h]: Don't think they're needed here either: https://github.com/cloudflare/workerd/pull/1914/commits/f8b58b5fa86d655035ca4b271362b9bcdb82c149

File: src/workerd/io/actor-cache.h
Comment (MellowYarker) [reply src/workerd/io/actor-cache.h:445]: What we really want is to prevent directly modifying the `syncStatus` unless it's part of some action (like adding to a list), but the methods that add to the respective lists don't have direct access to the `syncStatus`. The same issue would exist with `setClean` and `setDirty` -- though I suppose it would make it more obvious that you'd have to use `setNotInCache()` explicitly too...

**Review by a-robinson — APPROVED** [review]
Still LGTM, but it looks like merge on this will be blocked until @justin-mp removes his "Request changes" bit.

**Review by justin-mp — APPROVED** [review]

**Review by jasnell — APPROVED** [review]

**Pull request merged**
